lib/Recur/RRuleIterator.php

```diff
@@ -276,12 +284,60 @@ public function fastForward(\DateTimeInterface $dt): void
 
     /* Functions that advance the iterator {{{ */
 
+    /**
+     * Advances currentDate by the interval.
+     * Takes into account the case where summer time starts and
+     * the event time on that day may have had to be advanced,
+     * usually by 1 hour.
+     */
+    protected function advanceTheDate(string $interval): void
+    {
+        $hourOfPreviousDate = (int) $this->currentDate->format('G');
+        $this->currentDate = $this->currentDate->modify($interval);
```

Comment:
another approach could be to always add the time of the startdate to the interval string here:

```php
$this->currentDate = $this->currentDate->modify($interval . ' ' . $this->startDate->format('H:i:s'));
```

see https://3v4l.org/YKgAd

Reply:
Interesting, I didn't know that such an input was possible for `DateTime::modify()`

Reply:
I refactored - this time format thing means the hourJump logic is not needed here. So the code gets much simpler.

https://www.php.net/manual/en/datetime.formats.php
"Relative statements are always processed after non-relative statements. This makes "+1 week july 2008" and "july 2008 +1 week" equivalent."

I think that actually the interval "+2 days 02:30:00" gets processed as "set the time part to "02:30:00" then advance 2 days.

I can write the code:
```
$this->currentDate = $this->currentDate->modify( $this->startDate->format('H:i:s').' '.$interval);
```
And the tests still pass.

I'm not sure which is easiest for someone to understand.